frontend/__tests__/hooks/usePlaygroundActions.test.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import { renderHook, act } from "@testing-library/react";
+import { usePlaygroundActions } from "../../src/hooks/usePlaygroundActions";
+import { PlaygroundProvider } from "../../src/context/PlaygroundContext";
+
+// Mock uuid to return predictable values
+jest.mock("uuid", () => ({
+  v4: jest.fn(() => "test-uuid"),
+}));
+
+describe("usePlaygroundActions", () => {
+  // Wrap the hook in the PlaygroundProvider for testing
+  const wrapper = ({ children }: { children: React.ReactNode }) => (
+    <PlaygroundProvider>{children}</PlaygroundProvider>
+  );
+
+  beforeEach(() => {
+    // Clear mocks before each test
+    jest.clearAllMocks();
+  });
+
+  describe("addInputDocument", () => {
+    it("should create a new input document with the correct structure", () => {
+      const { result } = renderHook(() => usePlaygroundActions(), { wrapper });
```

Comment:
If this is repeated on each test can we move it into before each?

Reply:
I've been trying to figure out a way to do this and it doesn't seem supported to create something in the `beforeEach` and then "give" it to each test. Discussion about it here: https://github.com/jestjs/jest/issues/4903. Were you thinking of some specific implementation? I could do a fixture, but I'm not sure that saves us much code-wise.

Reply:
This isn't that big of a deal for these tests since its a single duplicated line on each test.